### PR TITLE
SPLICE-1270 Fixing Dictionary Index Lookup Logic

### PR DESCRIPTION
**Pull request by jleach4**
Inside the dictionary (not query stack), we currently do our index lookups from an index scan serially .  This will perform a single batch multi-get to reduce RPC calls.

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/base/SpliceController.java [hunk 1]
```diff
@@ -167,6 +185,51 @@ public boolean fetch(RowLocation loc,DataValueDescriptor[] destRow,FormatableBit
...
+                row.setRowArray(destRow);
+                row.resetRowArray();
+                DataCell keyValue=result.userData();
+                rowDecoder.set(keyValue.valueArray(),keyValue.valueOffset(),keyValue.valueLength());
+                rowDecoder.decode(row);
+                i++;
+            }
+            return true;
+        }catch(Exception e){
+            e.printStackTrace();
```
Comment (scottfines) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/base/SpliceController.java]: lose the extra printStackTrace() please

Comment (jleach4) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/base/SpliceController.java]: Oops...

Comment (jleach4) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/base/SpliceController.java]: Will fix and push, thanks.

Comment (jleach4) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/base/SpliceController.java]: Fixed

**Pull request merged**